tensorflow/core/kernels/inplace_ops.cc

```diff
@@ -46,7 +47,7 @@ Status DoParallelConcat(const CPUDevice& d, const Tensor& value, int32 loc,
   switch (value.dtype()) {
 #define CASE(type)                  \
   case DataTypeToEnum<type>::value: \
-    return DoParallelConcatUpdate<type>(d, value, loc, output);
+    return DoParallelConcatUpdate<CPUDevice, type>(d, value, loc, output);
```

Comment:
Shouldn't that be Device? Otherwise all DoParallelConcatUpdate will use CPUDevice